commands/src/main/java/org/wildfly/extras/creaper/commands/foundation/online/SnapshotBackup.java

```diff
@@ -0,0 +1,104 @@
+package org.wildfly.extras.creaper.commands.foundation.online;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.jboss.logging.Logger;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.admin.ReloadToSnapshot;
+
+import com.google.common.base.Strings;
+
+/**
+ * Provides a pair of online commands to backup and then restore the snapshot.
+ * <p>
+ * The {@code backup} command must be applied before {@code restore}, and
+ * {@code backup} can be applied once. If any one of these rules is violated, an
+ * exception is thrown. For special circumstances, when the backup that was
+ * already acquired is no longer needed and is not going to be restored, a
+ * {@code destroy} command is provided. If there was no backup acquired, the
+ * destroy command does nothing.
+ * </p>
+ */
+
+public final class SnapshotBackup {
+    private static final Logger log = Logger.getLogger(SnapshotBackup.class);
+
+    private String snapshotName; // null <=> backup wasn't acquired, can't restore
+
+    private final OnlineCommand backupPart = new OnlineCommand() {
+        @Override
+        public void apply(OnlineCommandContext occ) throws CommandFailedException, IOException, CliException {
+            if (SnapshotBackup.this.snapshotName != null) {
+                throw new CommandFailedException("Snapshot was already backed up as "
+                        + SnapshotBackup.this.snapshotName);
+            }
+            String snapshot = "";
+            ModelNodeResult modelNodeResult = occ.client.execute(":take-snapshot");
+            if (modelNodeResult.isSuccess() && modelNodeResult.hasDefinedValue()) {
+                snapshot = modelNodeResult.stringValue();
+            }
+            // returns absolute path, but we need just the name of snapshot
+            snapshot = snapshot.substring(snapshot.lastIndexOf(File.separator) + 1);
```

Comment:
Well file names on Windows can contain neither `/` nor `\`. So if the string contains `/`, it obviously isn't Windows and you can split on `/`, otherwise you split on `\`.

Re `AddModule`: this command only makes sense when running on the same system as the server. That doesn't apply in this situation.